src/components/workspace/BuildProject/BuildProject.tsx

```diff
@@ -102,23 +102,29 @@ const BuildProject: FC<Props> = ({ projectId, contract, updateContract }) => {
 
   const contractsToDeploy = () => {
     return projectFiles
-      .filter((f) => {
-        const _fileExtension = getFileExtension(f.name || '');
-        return (
-          f.path.startsWith(`${activeProject?.path}/dist`) &&
-          ['abi'].includes(_fileExtension as string)
+      .filter((file) => {
+        const fileExtension = getFileExtension(file.name || '');
+        const isAbiFile =
+          file.path.startsWith(`${activeProject?.path}/dist`) &&
```

Comment:
Please encapsulate all the logic related to filenames into a separate file.

If there is another place in the code that would need to check if the file is an ABI-file, it would be found there.

The underlying problem is that file paths are usually untyped, and it's unsafe to work with them with String methods. While working with paths as if they were Strings is not avoidable (that's the OS interface to work with files after all), it's possible to at least give proper attention to changes of unsafe code, if it's encapsulated in a separate file.